ignite/utils.py

```diff
@@ -169,3 +171,32 @@ def manual_seed(seed: int) -> None:
         np.random.seed(seed)
     except ImportError:
         pass
+
+
+def deprecated(deprecated_in: str, removed_in: str = "", reasons: list = [], raise_exception: bool = False) -> Callable:
```

Comment:
Please use the appropriate generic Type for `list`, then you can make it more precise `List[str]` 

And is it intentional to use a mutable object as a default for `reasons`?

Reply:
Thanks @gruebel !

> Please use the appropriate generic Type for list, then you can make it more precise List[str]

Sure!

> And is it intentional to use a mutable object as a default for reasons?

Do you mean that we could use something like tuples, I don't think there is any issue with using tuples here. Should we shift to tuples, I did not know if one was better than the other

Reply:
if you use a mutable object like `list` here, then some funny surprises can happen. 

Here a good link for extra infos https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments